hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/MapIndexScanPhysicalRel.java

```diff
@@ -45,6 +45,7 @@
     private final List<QueryDataType> converterTypes;
     private final RexNode indexExp;
     private final RexNode remainderExp;
+    private final boolean descending;
```

Comment:
A single flag for the whole index means that we will not be able to support multi-direction scans in the future without breaking the backward compatibility. Perhaps it makes sense to use `List<Boolean>`, where every element is the direction of the scan of an individual component. Now they will be either all `true, true, ...` or `false, false, ...`. 
But in the future, we will be able to add the multidirectional scan by simply changing the relevant `Exec` class, and without breaking the protocol.

Reply:
I'm not sure that I got the idea how the `List<Boolean>` will help with the multidirectional scan. My understanding is that if we want to support `[a ASC, b DESC]` ordering on the index level, we will have to extend our `SORTED` index configuration and explicitly specify this collation so that the entries will be stored pre-sorted. Such index will also support `[a DESC, b ASC] `sorting. In both cases, we don't need a per component specification of the ordering. This is why I'm not sure we have to add the `List<Boolean>`. @devozerov , can you please elaborate more on the idea?

Reply:
Consider that we have an index `[a ASC, b ASC]`. In theory, it should be possible to use it to do `ORDER BY a ASC, b DESC`. We do not support it now, but it should not be difficult to add such capability to our indexing infrastructure. If we add such a feature, it would not be possible to describe the desired direction with a single flag. Instead, we will pass `a ASC, b DESC`, which could be encoded as `List {false, true}`.

Reply:
Added in https://github.com/hazelcast/hazelcast/pull/18013/commits/858e478311e476afec0695dfe83552ce1fcff46f